lib/player.js

```diff
@@ -3646,6 +3648,39 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
           'streaming.preferNativeHls instead.');
     }
 
+    // map liveSyncMinLatency and liveSyncMaxLatency to liveSyncTargetLatency
+    // if liveSyncTargetLatency isn't set.
+    if (config['streaming'] &&
+      !('liveSyncTargetLatency' in config['streaming']) &&
+      ('liveSyncMinLatency' in config['streaming'] ||
+       'liveSyncMaxLatency' in config['streaming'])) {
+      const min = config['streaming']['liveSyncMinLatency'] ||
+          defaultConfig['streaming']['liveSyncMinLatency'];
+      const max = config['streaming']['liveSyncMaxLatency'] ||
+          defaultConfig['streaming']['liveSyncMaxLatency'];
+      const mid = Math.abs(max - min) / 2;
+      config['streaming']['liveSyncTargetLatency'] = min + mid;
+      config['streaming']['liveSyncTargetLatencyTolerance'] = mid;
+    }
+    // Deprecate 'streaming.liveSyncMaxLatency' configuration.
+    if (config['streaming'] && 'liveSyncMaxLatency' in config['streaming']) {
+      shaka.Deprecate.deprecateFeature(5,
+          'streaming.liveSyncMaxLatency',
+          'Please Use streaming.liveSyncTargetLatency and ' +
+          'streaming.liveSyncTargetLatencyTolerance instead. ' +
+          'Or, set the values in your DASH manifest');
+      delete config['streaming']['liveSyncMaxLatency'];
```

Comment:
Just like we have done with sourceBufferExtraFeatures (for example) you should remove the documentation from the extern, and you should also add all this information to `/docs/tutorials/upgrade.md`